spring-ai-core/src/main/java/org/springframework/ai/model/observation/ErrorLoggingObservationHandler.java

```diff
@@ -62,12 +64,13 @@ public ErrorLoggingObservationHandler(Tracer tracer,
 	}
 
 	@Override
-	public boolean supportsContext(Context context) {
-		return (context == null) ? false : this.supportedContextTypes.stream().anyMatch(clz -> clz.isInstance(context));
+	public boolean supportsContext(@Nullable Context context) {
```

Comment:
The `Context` is never null as per the Micrometer infrastructure. Is there a reason for making it nullable?

Reply:
I am going to remove the @Nullable annotation so it generates the appropriate warnings in the supported IDEs.